src/QueryPipeline/RemoteQueryExecutor.cpp

```diff
@@ -418,7 +418,7 @@ void RemoteQueryExecutor::sendQueryUnlocked(ClientInfo::QueryKind query_kind, As
 
     // Collect all roles granted on this node and pass those to the remote node
     std::vector<String> local_granted_roles;
-    if (context->getSettingsRef()[Setting::push_external_roles_in_interserver_queries] && !modified_client_info.initial_user.empty())
+    if (context->getSettingsRef()[Setting::push_external_roles_in_interserver_queries])
```

Comment:
In next line gets access control for user `modified_client_info.initial_user`. Is it works correct with empty user?

Reply:
the function that is called is `IAccessStorage::read(const std::vector<UUID> & ids, bool throw_if_not_exists)`. So it does not throw. Instead we have a check 2 lines below:
```cpp
if (user)
{
    ...
}
```

So everything is handled correctly